### PR TITLE
Fix links, typos, formatting in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by brgnepal**
Include fixes:
- Typo error
- Formatting links properly
- Superscript in necessary place
- Sentence formatting

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,16 +1,15 @@
 # Contributing to vagrant-service-manager Plugin
 
 The following is a set of guidelines for contributing to the
-vagrant-service-manager plugin, which is hosted in the `Project Atomic
-Organization <https://github.com/projectatomic>`_ on GitHub.
+`vagrant-service-manager` plugin, which is hosted in the [`Project Atomic
+Organization`](https://github.com/projectatomic) on GitHub.
 
-These are just guidelines, please use your best judgment and feel free
+These are just guidelines, please use your best judgement and feel free
```
Comment (bexelbie) [inline CONTRIBUTING.md:7]: How very OED of you :P

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
 to propose changes to this document in a pull request.
 
-At this point, this document is not complete, but
-as decisions are made on `container-tools@redhat.com
-<https://www.redhat.com/mailman/listinfo/container-tools>`_ mailing list
-they will be added to this document.
+At this point, this document is not complete, but as decisions are made on
```
Comment (bexelbie) [inline CONTRIBUTING.md]: At this point, this document is not complete, but as decisions are made on the

**Comment by bexelbie** [comment]
1 missing word, otherwise LGTM

**Comment by brgnepal** [comment]
@bexelbie Done :smile:

File: CONTRIBUTING.md (hunk 1 above)
Comment (brgnepal) [inline CONTRIBUTING.md:7]: :smile:

**Comment by navidshaikh** [comment]
LGTM

**Pull request merged**